### PR TITLE
jcli: allow to provide a custom TLS root certificate for REST API

### PR DESCRIPTION
**Pull request by eugene-babichenko**

### CODE REVIEW

File: jcli/src/jcli_app/rest/v0/diagnostic/mod.rs [hunk 1]
```diff
@@ -11,17 +11,19 @@ pub enum Diagnostic {
         addr: HostAddr,
         #[structopt(flatten)]
         debug: DebugFlag,
+        #[structopt(flatten)]
+        tls: TlsCert,
```
Comment (mzabaluev) [inline jcli/src/jcli_app/rest/v0/diagnostic/mod.rs:15]: These options are getting a bit repetitive, maybe there should be a struct for them that will be flattened into commands' options.

Comment (eugene-babichenko) [reply jcli/src/jcli_app/rest/v0/diagnostic/mod.rs:15]: Actually, I thought about making a single structure out of `addr, `debug` and `tls`, that will have a method to build a client perform a request.

**Comment by dkijania** [comment]
hmm, got an issue when running command:

```
jcli rest v0 node stats get --host http://127.0.0.1:8562/api --tls-cert-path C:\Work\iohk\pki\ca.crt
failed to make a REST request
  |-> communication with node failed in unexpected way
  |-> error sending request for url (http://127.0.0.1:8562/api/v0/node/stats): connection closed before message completed
  |-> connection closed before message completed
```

while my certs are:
[tls.zip](https://github.com/input-output-hk/jormungandr/files/5577440/tls.zip)

**Comment by eugene-babichenko** [comment]
Probably it's because you have http and https in your request. I tested your certs and they work just fine.

**Review by mzabaluev — APPROVED** [review]

**Pull request merged**